cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/http/SystemHttpServiceContext.java

```diff
@@ -61,4 +61,15 @@
    */
   Map<String, String> evaluateMacros(String namespace, Map<String, String> properties,
                                      MacroEvaluator evaluator, MacroParserOptions options) throws InvalidMacroException;
+
+  /**
+   * Get preferences for the given entity
```

Comment:
Should add more detail here, like how this is a live call that fetches the resolved preferences at the current point in time, and is different from getRuntimeArguments(), which gets arguments as they were at the start of the program run.